drivers/overlay/ov_network.go

```diff
@@ -393,10 +393,12 @@ func isOverlap(nw *net.IPNet) bool {
 	}
 
 	if err := netutils.CheckNameserverOverlaps(nameservers, nw); err != nil {
+		logrus.Errorf("overlay subnet %s conflicts with nameservers %v", nw.String(), nameservers)
```

Comment:
Instead of adding logs, why don't we modify this function to return some kind of overlap type and then return that as part of error from the caller? That way it will show up in the error response rather than in the logs.